src/azure-cli-core/azure/cli/core/profiles/_shared.py

```diff
@@ -216,11 +242,38 @@ def __lt__(self, other):  # pylint: disable=too-many-return-statements
         return False
 
 
+def _parse_api_version(api_version):
+    """Will try to parse it as a date, and if not working
+    as semver, and if still not working raise.
+    """
+    try:
+        return _DateAPIFormat(api_version)
+    except ValueError:
+        return _SemVerAPIFormat(api_version)
+
+
+def _cross_api_format_lt(api_version, other):
+    """LT strategy that supports if types are different.
+
+    For now, let's assume that any Semver is higher than any DateAPI
+    This fits KeyVault, if later we have a counter-example we'll update
+    """
+    api_version = _parse_api_version(api_version)
+    other = _parse_api_version(other)
+
+    if type(api_version) is type(other):
+        return api_version < other
+    elif isinstance(api_version, _DateAPIFormat) and isinstance(other, _SemVerAPIFormat):
+        return True
+    return False
```

Comment:
this and 3 lines above can be simplified as: `return isinstance(api_version, _DateAPIFormat) and isinstance(other, _SemVerAPIFormat)`

Reply:
Yes, this one is more for clarity. But that's technically true.